# 📝 Examen de Pandas Básico


## Introducción

Una empresa multinacional dedicada a la comercialización de frutas frescas opera en distintos mercados a través de sucursales físicas y canales digitales. Diariamente, la compañía registra miles de transacciones provenientes de diversas zonas geográficas y de su plataforma online.

Debido al crecimiento del negocio y al aumento en el volumen de datos, la empresa ha identificado la necesidad de mejorar su capacidad de análisis para entender mejor el comportamiento de sus ventas. En particular, busca responder preguntas clave como: qué productos generan mayores ingresos, qué zonas concentran más ventas y cómo se distribuye la demanda entre sus distintos clientes.

Para ello, el equipo de datos ha dispuesto un archivo llamado “Ventas.csv”, que contiene el detalle de las transacciones realizadas durante un día específico. Este archivo incluye información sobre la zona de venta, el cliente, el producto, la cantidad vendida y el precio unitario.

Tu rol como analista es procesar esta información utilizando Python y Pandas, con el objetivo de generar indicadores claros y un resumen que apoye la toma de decisiones comerciales.


**Archivo de referencia:** `Ventas.csv`

**Columnas:**  
`Numero,Zona,Cliente,Producto,Cantidad,Precio`

---

## Instrucciones
1. Usar Python y Pandas para resolver todas las preguntas.  
2. Guardar los resultados en variables y mostrar con `print()` cuando corresponda.  

---

# Preguntas

## 1️⃣ Leer el archivo
- Importa Pandas y carga el archivo `Ventas.csv` en un DataFrame llamado `df`.
- Muestra las primeras 7 filas.

In [1]:
#Importar librerías necesarias
import pandas as pd

In [2]:
#Cargar el archivo CSV en un DataFrame
df = pd.read_csv('data/raw/Ventas.csv')

#Mostrar las primeras 7 filas del DataFrame
primeras_7_filas = df.head(7)
print(primeras_7_filas)

   Numero    Zona                 Cliente Producto  Cantidad  Precio
0       1   Norte          UrbanSphere Co  Naranja        52       8
1       2  Online      EcoWave Industries  Naranja       196       8
2       3  Centro  GreenField Enterprises      Uva       435       7
3       4   Norte          Orion Ventures    Mango       641       4
4       5     Sur          NovaLink Group     Pera        48      10
5       6     Sur   ClearStream Solutions     Pera       313      10
6       7  Online       Infinity Networks   Cereza       991       2


## 2️⃣ Inspección

- Cuantas filas y columnas tiene el Dataframe.

In [3]:
#Calcular y mostrar numero e filas con shape
numero_filas = df.shape[0]
print(f'Número de filas: {numero_filas}')

#Calcular y mostrar numero de columnas con shape
numero_columnas = df.shape[1]
print(f'Número de columnas: {numero_columnas}')

Número de filas: 1000
Número de columnas: 6


## 3️⃣ Selección de columnas

- Muestra solo las columnas Cliente y Producto.

In [4]:
#Crear nueva df con solo las columnas Cliente y Producto
df_cliente_producto = df[['Cliente', 'Producto']]
print(df_cliente_producto)

                    Cliente Producto
0            UrbanSphere Co  Naranja
1        EcoWave Industries  Naranja
2    GreenField Enterprises      Uva
3            Orion Ventures    Mango
4            NovaLink Group     Pera
..                      ...      ...
995        Pioneer Robotics  Manzana
996         SummitWorks Inc     Piña
997          Orion Ventures  Naranja
998      TechNova Solutions    Fresa
999       AeroGlide Systems   Cereza

[1000 rows x 2 columns]


## 4️⃣ Filtrado de datos

- Muestra todas las ventas que se hicieron en la zona "Norte".

In [5]:
#Calcular y mostrar solo las ventas de la zona Norte
ventas_zona_norte = df[df['Zona'] == 'Norte']
print(ventas_zona_norte)

     Numero   Zona                Cliente Producto  Cantidad  Precio
0         1  Norte         UrbanSphere Co  Naranja        52       8
3         4  Norte         Orion Ventures    Mango       641       4
8         9  Norte      Infinity Networks   Banana        21       9
12       13  Norte     TechNova Solutions    Mango       604       4
13       14  Norte      Infinity Networks    Fresa       230       2
..      ...    ...                    ...      ...       ...     ...
954     955  Norte  BluePeak Technologies      Uva       754       7
975     976  Norte         Orion Ventures  Manzana       454       2
979     980  Norte     EcoWave Industries  Manzana        50       2
985     986  Norte  BluePeak Technologies   Banana       488       9
987     988  Norte       Skyline Dynamics    Fresa       370       2

[234 rows x 6 columns]


## 5️⃣ Operaciones Básicas

- Crea una nueva columna Total con el cálculo de (Cantidad * Precio).
- Muestra el DataFrame actualizado ( 8 primeras filas)

In [6]:
#Crear nueva columna Total multiplicando Cantidad por Precio
df['Total'] = df['Cantidad'] * df['Precio']

#Guardar en variable las primeras 7 filas del DataFrame con la nueva columna Total
primeras_7_filas_con_total = df.head(7)

#Mostrar las primeras 7 filas del DataFrame con la nueva columna Total
print(primeras_7_filas_con_total)

   Numero    Zona                 Cliente Producto  Cantidad  Precio  Total
0       1   Norte          UrbanSphere Co  Naranja        52       8    416
1       2  Online      EcoWave Industries  Naranja       196       8   1568
2       3  Centro  GreenField Enterprises      Uva       435       7   3045
3       4   Norte          Orion Ventures    Mango       641       4   2564
4       5     Sur          NovaLink Group     Pera        48      10    480
5       6     Sur   ClearStream Solutions     Pera       313      10   3130
6       7  Online       Infinity Networks   Cereza       991       2   1982


## 6️⃣ Análisis

- ¿Cuál fué el total de ventas del día?
- ¿Cuál es la venta por cada zona?

In [7]:
#Calcular y mostrar total de ventas del día
total_ventas_dia = df['Total'].sum()

print(f'Total de ventas del día: {total_ventas_dia}')
print('-' * 40)

#Calcular y mostrar total de ventas por zona
total_ventas_por_zona = df.groupby('Zona')['Total'].sum()

print(f'Total de ventas por zona:\n{total_ventas_por_zona}')

Total de ventas del día: 2702306
----------------------------------------
Total de ventas por zona:
Zona
Centro    711993
Norte     644954
Online    720846
Sur       624513
Name: Total, dtype: int64


## 7️⃣ Guardar Resultados

- Generar un archivo "resumen.csv" que contenga:

    - Suma de Ventas por Zona (Descripcion y Total)
    - Suma de Ventas por producto (Descripcion y Total)
    - Cantidad de registros (transacciones) : xxxxx
    - Suma de Ventas: yyyyy

- El archivo debiera tener 3 columnas:
    - Descripcion 
    - Valor
    - Tipo

- Ejemplo:


| Descripción | Valor  | Tipo      |
|------------|-----------|-----------|
| Centro     | 711,993   | Zonas     |
...
| Sur        | 624,513   | Zonas     |
| Banana     | 429,714   | Productos |
| Cereza     | 107,314   | Productos |
...
| Piña       | 335,352   | Productos |
| Uva        | 352,107   | Productos |
| venta      | 2,702,306 | Resumen   |
| registros  | 1,000     | Resumen   |



Ayuda: Algunas maneras de renombrar Las columnas de un dataframe:

- df = df.rename(columns={'nombre_antiguo': 'nombre_nuevo'})
- df.rename(columns={"A": "Columna1", "B": "Columna2"}, inplace=True)
- df.columns = ['nuevo_nombre1', 'nuevo_nombre2', ...]

In [8]:
#Crear variables resumen_zonas, resumen_productos y resumen_general con los totales de ventas por zona, producto y un resumen general respectivamente. Luego concatenar estas variables en una sola variable resumen y formatear la columna Valor para mostrar los valores con formato de moneda. Finalmente, guardar el resultado en un archivo CSV llamado resumen.csv y mostrar el resultado.
resumen_zonas = df.groupby('Zona')['Total'].sum().reset_index()
resumen_zonas.columns = ['Descripcion', 'Valor']
resumen_zonas['Tipo'] = 'Zonas'

resumen_productos = df.groupby('Producto')['Total'].sum().reset_index()
resumen_productos.columns = ['Descripcion', 'Valor']
resumen_productos['Tipo'] = 'Productos'

resumen_general = pd.DataFrame({
    'Descripcion': ['venta', 'registros'],
    'Valor': [df['Total'].sum(), len(df)],
    'Tipo': ['Resumen', 'Resumen']
})

# Concatenar los resúmenes y formatear la columna Valor
# Se usó IA paara ayudar a formatear la columna valor.

resumen = pd.concat([resumen_zonas, resumen_productos, resumen_general], ignore_index=True)
resumen['Valor'] = resumen.apply(
    lambda fila: f"${fila['Valor']:,.0f}".replace(',', '.') if fila['Descripcion'] != 'registros' else f"{fila['Valor']:,.0f}".replace(',', '.'),
    axis=1
)

# Guardar el resultado en un archivo CSV llamado resumen.csv
resumen.to_csv('data/output/resumen.csv', index=False)

# Mostrar el resultado
print(resumen)

   Descripcion       Valor       Tipo
0       Centro    $711.993      Zonas
1        Norte    $644.954      Zonas
2       Online    $720.846      Zonas
3          Sur    $624.513      Zonas
4       Banana    $429.714  Productos
5       Cereza    $107.314  Productos
6        Fresa    $105.436  Productos
7         Kiwi    $166.965  Productos
8        Mango    $169.372  Productos
9      Manzana     $88.136  Productos
10     Naranja    $420.880  Productos
11        Pera    $527.030  Productos
12        Piña    $335.352  Productos
13         Uva    $352.107  Productos
14       venta  $2.702.306    Resumen
15   registros       1.000    Resumen
